tensorflow/contrib/tensorrt/kernels/trt_engine_op.cc

```diff
@@ -402,9 +402,12 @@ bool TRTEngineOp::ExecuteTrtEngine(
                                                 ->implementation()
                                                 ->GpuStreamMemberHack()));
 
+  // nvinfer1::IExecutionContext::enqueue is not thread safe and we need a mutex
```

Comment:
I would rephrase this to the following:

nvinfer1::IExecutionContext::enqueue can be used by multiple threads simultaneously as long different threads use different execution contexts. But since we use one execution context for all threads, we guard the runtime by a mutex. Since nvinfer1::IExecutionContext::enqueue is an asynchronous function, this guard will be safe only if all the threads use the same CUDA stream which is the current behavior in Tensorflow.

Reply:
Thanks. Since this is a comment change, instead of added a separate commit for that to the PR, I'll fix it in another commit to head.